operators/src/processing/raster_scaling.rs

```diff
@@ -0,0 +1,358 @@
+            scale_with: self.params.scale_with,
+            offset_by: self.params.offset_by,
+            result_descriptor: out_desc,
+            source: input,
+            scaling_mode: self.params.scaling_mode,
+        };
+
+        Ok(initialized_operator.boxed())
+    }
+}
+
+impl InitializedRasterOperator for InitializedRasterScalingOperator {
+    fn result_descriptor(&self) -> &RasterResultDescriptor {
+        &self.result_descriptor
+    }
+
+    fn query_processor(&self) -> Result<TypedRasterQueryProcessor> {
+        let scale_with = self.scale_with.clone();
+        let offset_by = self.offset_by.clone();
+        let no_data_value = self.result_descriptor.no_data_value;
+        let source = self.source.query_processor()?;
+        let scaling_mode = self.scaling_mode;
+
+        let res_op = call_on_generic_raster_processor!(source, source_proc => { TypedRasterQueryProcessor::from(RasterScalingProcessor::create_boxed(scale_with, offset_by,  source_proc, no_data_value, scaling_mode)) });
+
+        Ok(res_op)
+    }
+}
+
+struct RasterScalingProcessor<Q, P>
+where
+    Q: RasterQueryProcessor<RasterType = P>,
+{
+    source: Q,
+    scale_with: PropertiesKeyOrValue,
+    offset_by: PropertiesKeyOrValue,
+    no_data_value: P,
+    scaling_mode: ScalingMode,
+}
+
+impl<Q, P> RasterScalingProcessor<Q, P>
+where
+    Q: RasterQueryProcessor<RasterType = P> + 'static,
+    P: Pixel + Scale + Unscale + FromPrimitive,
+    f64: AsPrimitive<P>,
+{
+    pub fn create_boxed(
+        scale_with: PropertiesKeyOrValue,
+        offset_by: PropertiesKeyOrValue,
+        source: Q,
+        no_data_value: Option<f64>,
+        scaling_mode: ScalingMode,
+    ) -> Box<dyn RasterQueryProcessor<RasterType = P>> {
+        RasterScalingProcessor::create(
+            scale_with,
+            offset_by,
+            source,
+            no_data_value.map_or(P::DEFAULT_NO_DATA_VALUE, num_traits::AsPrimitive::as_),
+            scaling_mode,
+        )
+        .boxed()
+    }
+
+    pub fn create(
+        scale_with: PropertiesKeyOrValue,
+        offset_by: PropertiesKeyOrValue,
+        source: Q,
+        no_data_value: P,
+        scaling_mode: ScalingMode,
+    ) -> RasterScalingProcessor<Q, P> {
+        RasterScalingProcessor {
+            source,
+            scale_with,
+            offset_by,
+            no_data_value,
+            scaling_mode,
+        }
+    }
+
+    async fn scale_tile_async(
+        &self,
+        tile: RasterTile2D<P>,
+        pool: Arc<ThreadPool>,
+    ) -> Result<RasterTile2D<P>> {
+        let offset_by = Self::prop_value(&self.offset_by, &tile.properties)?;
+        let scale_with = Self::prop_value(&self.scale_with, &tile.properties)?;
+        let scaling_mode = self.scaling_mode;
+
+        let no_data_value = self.no_data_value;
+
+        let res_tile =
+            crate::util::spawn_blocking_with_thread_pool(pool, move || match scaling_mode {
+                ScalingMode::Scale => {
```

Comment:
could maybe also link this statically by making the processor generic over some T with a static scale/unscale method.

Reply:
[f0e8659](https://github.com/geo-engine/geoengine/pull/530/commits/f0e86596f91cdee8832675c8fa8f4c7546aede1a)